New Cycamore Enrichment Facility

mbmcgarry opened the pull request:
This PR is dependent on cyclus/cyclus#1084 and cyclus/cyclus#1093

rwcarlsen commented:
I also have a fix/improvement coming for cycpp that will require the custom
schema snipped to be modified (simplified).  So this should probably wait
for that too.

On Tue, Mar 10, 2015 at 12:19 PM, mbmcgarry <notifications@github.com>
wrote:

> This PR is dependent on cyclus/cyclus#1084
> <https://github.com/cyclus/cyclus/pull/1084> and cyclus/cyclus#1093
> <https://github.com/cyclus/cyclus/pull/1093>
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/cyclus/cycamore/pull/305
> Commit Summary
>
>    - first cut at checking EF input composition
>    - added check for multiple isotopes, still working on tails buffer
>    - check isotope composition and implicit FeedAssay calculation
>    - draft tails buffer
>    - more work on tails buffer
>    - tails buffer works
>    - working on max_enrichment state var
>    - minor fix to max_enrich
>    - modified original input files
>    - typo
>    - removed sqlite file
>    - Merge branch 'sink' of https://github.com/mbmcgarry/cycamore into
>    new_EF
>    - working on custom pragma for enrichment range
>    - bug in material pointer
>    - Merge branch 'develop' into new_EF
>    - fixed FeedQty so that non-fissile materials are properly accounted
>    for
>    - changed NonFissileMultiplier to be a toolkit fn
>    - not-working version of limits on max-enrichment
>    - minor - changed float to double
>    - max_enrich schema still not working
>    - better schema formatting but parser still fails
>    - working on trade prefs
>    - accidentally missed this change yesterday
>    - fixed xml syntax, and everything compiles. Still some logic bugs
>    though.
>    - preference sorting works
>    - removed NonFissileMultiplier function calls and replaced with
>    multi_mass_frac. Fixed bug in preference-setting for zero mass. Started
>    documentation updates.
>    - documentation
>    - flag for preference ordering
>    - added tests for new features. Some old tests are still broken
>    - more test editing
>    - almost finished with tests
>    - style guide cleanup
>    - more cleanup
>    - another cleanup
>
> File Changes
>
>    - *M* input/enrichment/1_src_enr_rxtr_sink.xml
>    <https://github.com/cyclus/cycamore/pull/305/files#diff-0> (1)
>    - *M* input/enrichment/linear_src_enr_rxtr_sink.xml
>    <https://github.com/cyclus/cycamore/pull/305/files#diff-1> (3)
>    - *M* input/enrichment/natu_capacitated.xml
>    <https://github.com/cyclus/cycamore/pull/305/files#diff-2> (1)
>    - *M* input/enrichment/swu_capacitated.xml
>    <https://github.com/cyclus/cycamore/pull/305/files#diff-3> (1)
>    - *M* input/physor/1_Enrichment_2_Reactor.xml
>    <https://github.com/cyclus/cycamore/pull/305/files#diff-4> (1)
>    - *M* src/enrichment_facility.cc
>    <https://github.com/cyclus/cycamore/pull/305/files#diff-5> (287)
>    - *M* src/enrichment_facility.h
>    <https://github.com/cyclus/cycamore/pull/305/files#diff-6> (298)
>    - *M* src/enrichment_facility_tests.cc
>    <https://github.com/cyclus/cycamore/pull/305/files#diff-7> (733)
>    - *M* src/enrichment_facility_tests.h
>    <https://github.com/cyclus/cycamore/pull/305/files#diff-8> (6)
>    - *A* tests/run_inputs.py
>    <https://github.com/cyclus/cycamore/pull/305/files#diff-9> (153)
>
> Patch Links:
>
>    - https://github.com/cyclus/cycamore/pull/305.patch
>    - https://github.com/cyclus/cycamore/pull/305.diff
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cyclus/cycamore/pull/305>.
>

rwcarlsen commented:
s/snipped/snippet/

On Tue, Mar 10, 2015 at 12:34 PM, Robert Carlsen <rwcarlsen@gmail.com>
wrote:

> I also have a fix/improvement coming for cycpp that will require the
> custom schema snipped to be modified (simplified).  So this should probably
> wait for that too.
>
> On Tue, Mar 10, 2015 at 12:19 PM, mbmcgarry <notifications@github.com>
> wrote:
>
>> This PR is dependent on cyclus/cyclus#1084
>> <https://github.com/cyclus/cyclus/pull/1084> and cyclus/cyclus#1093
>> <https://github.com/cyclus/cyclus/pull/1093>
>> ------------------------------
>> You can view, comment on, or merge this pull request online at:
>>
>>   https://github.com/cyclus/cycamore/pull/305
>> Commit Summary
>>
>>    - first cut at checking EF input composition
>>    - added check for multiple isotopes, still working on tails buffer
>>    - check isotope composition and implicit FeedAssay calculation
>>    - draft tails buffer
>>    - more work on tails buffer
>>    - tails buffer works
>>    - working on max_enrichment state var
>>    - minor fix to max_enrich
>>    - modified original input files
>>    - typo
>>    - removed sqlite file
>>    - Merge branch 'sink' of https://github.com/mbmcgarry/cycamore into
>>    new_EF
>>    - working on custom pragma for enrichment range
>>    - bug in material pointer
>>    - Merge branch 'develop' into new_EF
>>    - fixed FeedQty so that non-fissile materials are properly accounted
>>    for
>>    - changed NonFissileMultiplier to be a toolkit fn
>>    - not-working version of limits on max-enrichment
>>    - minor - changed float to double
>>    - max_enrich schema still not working
>>    - better schema formatting but parser still fails
>>    - working on trade prefs
>>    - accidentally missed this change yesterday
>>    - fixed xml syntax, and everything compiles. Still some logic bugs
>>    though.
>>    - preference sorting works
>>    - removed NonFissileMultiplier function calls and replaced with
>>    multi_mass_frac. Fixed bug in preference-setting for zero mass. Started
>>    documentation updates.
>>    - documentation
>>    - flag for preference ordering
>>    - added tests for new features. Some old tests are still broken
>>    - more test editing
>>    - almost finished with tests
>>    - style guide cleanup
>>    - more cleanup
>>    - another cleanup
>>
>> File Changes
>>
>>    - *M* input/enrichment/1_src_enr_rxtr_sink.xml
>>    <https://github.com/cyclus/cycamore/pull/305/files#diff-0> (1)
>>    - *M* input/enrichment/linear_src_enr_rxtr_sink.xml
>>    <https://github.com/cyclus/cycamore/pull/305/files#diff-1> (3)
>>    - *M* input/enrichment/natu_capacitated.xml
>>    <https://github.com/cyclus/cycamore/pull/305/files#diff-2> (1)
>>    - *M* input/enrichment/swu_capacitated.xml
>>    <https://github.com/cyclus/cycamore/pull/305/files#diff-3> (1)
>>    - *M* input/physor/1_Enrichment_2_Reactor.xml
>>    <https://github.com/cyclus/cycamore/pull/305/files#diff-4> (1)
>>    - *M* src/enrichment_facility.cc
>>    <https://github.com/cyclus/cycamore/pull/305/files#diff-5> (287)
>>    - *M* src/enrichment_facility.h
>>    <https://github.com/cyclus/cycamore/pull/305/files#diff-6> (298)
>>    - *M* src/enrichment_facility_tests.cc
>>    <https://github.com/cyclus/cycamore/pull/305/files#diff-7> (733)
>>    - *M* src/enrichment_facility_tests.h
>>    <https://github.com/cyclus/cycamore/pull/305/files#diff-8> (6)
>>    - *A* tests/run_inputs.py
>>    <https://github.com/cyclus/cycamore/pull/305/files#diff-9> (153)
>>
>> Patch Links:
>>
>>    - https://github.com/cyclus/cycamore/pull/305.patch
>>    - https://github.com/cyclus/cycamore/pull/305.diff
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/cyclus/cycamore/pull/305>.
>>
>
>

rwcarlsen commented:
also blocked on cyclus/cyclus#1098

scopatz commented on tests/run_inputs.py:
```diff
@@ -0,0 +1,153 @@
+                    inputs.append(os.path.join(root, name))
+            else :
+                files.remove(name)
+    print "The catalogs to be moved are:"
```
Not Python 3 compatible.

mbmcgarry replied:
I'm confused about this tests/run_inputs.py.  This is not a file that I wrote, but it appears that it wasn't in the old EF - perhaps it is automatically generated and was supposed to be in .ignore?

scopatz replied:
Oh yeah, run_inputs is old.  Not sure what is the deal here....

scopatz commented on tests/run_inputs.py:
```diff
@@ -0,0 +1,153 @@
+    """This function walks the 'path' tree and finds input files"""
+    catalogs=[]
+    catalognames=[]
+    inputs=[]
```
PEP8 single spaces surrounding `=`

scopatz commented on tests/run_inputs.py:
```diff
@@ -0,0 +1,153 @@
+def copy_catalogs(catalogs,cyclus_path) :
+    """Copies files in the catalogs list to the cyclus executable directory""" 
+    for cat in catalogs : 
+        p = Popen("cp "+cat+" "+cyclus_path,
```
use `check_call()` or something.  Also, is there a need to run this in shell mode? Also spaces around `+`

scopatz commented on tests/run_inputs.py:
```diff
@@ -0,0 +1,153 @@
+    def add_to_summary(self, test_file) :
+        """Adds a test file to this summary object"""
+        if test_file.passed : 
+            self.passed.append( test_file.infile )
```
PEP8 no spaces buffering parentheses.

scopatz commented on tests/run_inputs.py:
```diff
@@ -0,0 +1,153 @@
+        print "Input file " + self.infile 
+        if re.search("No such file or directory",output) :
+            print "Cyclus executable not found in path."
+        elif re.search("ERROR",output) or re.search("Segmentation fault",output):
```
PEP8: spaces after commas

mbmcgarry commented:
Ping - this PR is no longer blocked.

rwcarlsen commented on src/enrichment_facility.h:
```diff
@@ -72,70 +79,90 @@ class NatUConverter : public cyclus::Converter<cyclus::Material> {
+///  into a requested enriched recipe (ie. 4% enriched uranium), given
+///  the natural uranium inventory constraint and its SWU capacity
+///  constraint.
+
```
need triple slash on this line to maintain continuity of doc comment for doxygen.

rwcarlsen commented on src/enrichment_facility.h:
```diff
@@ -72,70 +79,90 @@ class NatUConverter : public cyclus::Converter<cyclus::Material> {
+///  U235 is present in the offered material, the trade preference is set
+///  to -1 and the material is not accepted.  Any material components other
+///  other than U235 and U238 are sent directly to the tails buffer.
+
```
same - triple slash

rwcarlsen commented on src/enrichment_facility.h:
```diff
@@ -72,70 +79,90 @@ class NatUConverter : public cyclus::Converter<cyclus::Material> {
+///  requested and the facility does not have the SWU or quantity capacity
+///  to meet all requests, the requests are fully, then partially filled
+///  in unspecified but repeatable order.
+
```
triple.

mbmcgarry replied:
fixed 3 triple-slash omissions.

rwcarlsen commented on src/enrichment_facility.h:
```diff
@@ -72,70 +79,90 @@ class NatUConverter : public cyclus::Converter<cyclus::Material> {
+  "in unspecified but repeatable order."				\
+  "\n\n"								\
+  "EnrichmentFacility also offers its tails as an output commodity with" \
+  "no associated recipe.  Bids for tails are constrained only by total" \
```
"Recipe" used in this context is a bit strange.  The offerred tails material has a composition - and we offer the actual composition that we will be trading away.  So I think something like:

```
Accumulated tails inventory is offered for trading on a specifiable output commodity.
```

No need to mention constraints - that is implicit - the enrichment facility can't ever trade away more tails than it has :-)

mbmcgarry replied:
done.

rwcarlsen commented on src/enrichment_facility.h:
```diff
@@ -233,81 +266,126 @@ class EnrichmentFacility : public cyclus::Facility {
+    "doc": "commodity that the enrichment facility accepts",		\
+    "uitype": "incommodity" \
+  }
   std::string in_commod;
```
rename to `feed`

rwcarlsen commented on src/enrichment_facility.h:
```diff
@@ -233,81 +266,126 @@ class EnrichmentFacility : public cyclus::Facility {
+    "doc": "commodity that the enrichment facility supplies",		\
+    "uitype": "outcommodity" \
+  }
   std::string out_commod;
```
rename to `product` or `enriched`

rwcarlsen commented on src/enrichment_facility.h:
```diff
@@ -233,81 +266,126 @@ class EnrichmentFacility : public cyclus::Facility {
+    "doc": "recipe for enrichment facility input commodity",		\
+    "uitype": "recipe" \
+  }
   std::string in_recipe;
```
`feed_recipe`

rwcarlsen commented on src/enrichment_facility.h:
```diff
@@ -233,81 +266,126 @@ class EnrichmentFacility : public cyclus::Facility {
+    "doc": "amount of natural uranium stored at the enrichment "       \
+           "facility at the beginning of the simulation (kg)"		\
+  }
   double initial_reserves;
```
`initial_feed`

gidden commented:
I'm +1 for the name renames, but note that they will require input file updates

rwcarlsen commented on src/enrichment_facility.h:
```diff
@@ -233,81 +266,126 @@ class EnrichmentFacility : public cyclus::Facility {
+    "doc": "maximum total inventory of natural uranium in "		\
+           "the enrichment facility (kg)" \
+  }
   double max_inv_size;
```
`max_feed_size`

mbmcgarry replied:
Changed the following variable names:
in_commod -> feed
in_commodity() -> feed_commod()
out_commod -> product
in_recipe -> feed_recipe
initial_reserves -> initial_feed
InitialReserves -> InitialFeed
max_inv_size -> max_feed_inventory

I prefer to keep 'inventory' rather than 'size' because that is the precise word to describe the amount of material stored at the facility.

rwcarlsen commented on src/enrichment_facility.h:
```diff
@@ -233,81 +266,126 @@ class EnrichmentFacility : public cyclus::Facility {
+           "so that EF chooses higher U235 content first" \
+  }
+  bool order_prefs;
+  #pragma cyclus var { 'derived_init': 'current_swu_capacity = swu_capacity;' }
```
You should remove this pragma.  `current_swu_capacity` is a intra-timestep state variable and is just used for temporary state between a bunch of function calls that occur on a single time step.  No need for it to be a managed state variable.  Also, it doesn't need a default value from the pragma because it gets set equal to swu_capacity at the beginning of every Tick anyway.

rwcarlsen commented:
I also recomment we rename EnrichmentFacility to not have the word "Facility" in it.  I'm okay with just `Enrichment` unless anyone has other suggestions.

gidden commented:
Because this also fabricates fuel, I would offer `EnrichmentFab` or `EnrFab` as other alternatives.

mbmcgarry commented:
Not yet ready - I still need to address this derived_var issue.

Unrelatedly, I have changed the name from EnrichmentFacility to Enrichment.  I decided against adding 'Fab' because although it can implicitly fabricate fuel, we now have an explicit FuelFab facility.  I find it less confusing this way.

mbmcgarry commented:
Ping - this is ready for review again. derived_var issue circumvented by the fact that didn't want the derived var to be tracked as a state var anyway.

gidden commented on src/enrichment_facility.h:
```diff
@@ -233,81 +265,127 @@ class EnrichmentFacility : public cyclus::Facility {
+    "doc": "feed commodity that the enrichment facility accepts",	\
+    "uitype": "incommodity" \
+  }
+  std::string feed;
```
Sorry to make you thrash here @mbmcgarry, but I think this should be `feed_commod` and below should be `product_commod`, much like we have `tails_commod`. Explicit being better than implicit. Input files will need updating as well.

mbmcgarry replied:
No problem. I was thinking about this myself as I was making the first round of changes. . .

gidden commented:
I am seeing a test failure:
```
.....F.......
======================================================================
FAIL: test_deploy (test_regression.TestPhysorEnrichment)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/gidden/work/cyclus/cycamore/tests/test_regression.py", line 76, in test_deploy
    assert_equal(len(self.enr_id), 1)
AssertionError: 0 != 1
-------------------- >> begin captured stdout << ---------------------
TESTING: running command in /home/gidden/work/cyclus/cycamore/tests:

cyclus -o 5b07836e-76fb-45fe-8446-9c4a9050938a.h5 --input-file ../input/physor/1_Enrichment_2_Reactor.xml


--------------------- >> end captured stdout << ----------------------

----------------------------------------------------------------------
Ran 13 tests in 2.057s

FAILED (failures=1)
```

gidden commented on src/enrichment_facility.cc:
```diff
@@ -87,9 +90,72 @@ EnrichmentFacility::GetMatlRequests() {
-void EnrichmentFacility::AcceptMatlTrades(
-  const std::vector< std::pair<cyclus::Trade<cyclus::Material>,
-  cyclus::Material::Ptr> >& responses) {
+// Sort bids by U-235 content
```
kill this comment, please (or move it to the header file as needed)

gidden commented:
@mbmcgarry, can you provide a short synopsis of the substantive changes this is adding to the archetype? It will help my review to have a big-picture view of it.

gidden commented on src/enrichment_facility.cc:
```diff
@@ -98,11 +164,11 @@ void EnrichmentFacility::AcceptMatlTrades(
 std::set<cyclus::BidPortfolio<cyclus::Material>::Ptr>
-EnrichmentFacility::GetMatlBids(
-  cyclus::CommodMap<cyclus::Material>::type& commod_requests) {
+    Enrichment::GetMatlBids(
```
no indent

gidden commented on src/enrichment_facility.h:
```diff
@@ -57,11 +57,18 @@ class NatUConverter : public cyclus::Converter<cyclus::Material> {
+
+    double natu_frac = mq.mass_frac(nucs);
+    double natu_req = cyclus::toolkit::FeedQty(m->quantity(), assays);
+    return natu_req/natu_frac;
```
space between division operator

gidden commented on src/enrichment_facility.h:
```diff
@@ -57,11 +57,18 @@ class NatUConverter : public cyclus::Converter<cyclus::Material> {
 
   /// @returns true if Converter is a NatUConverter and feed and tails equal
-  virtual bool operator==(Converter& other) const {
+  virtual bool operator == (Converter& other) const {
```
no space

gidden commented on src/enrichment_facility.h:
```diff
@@ -186,40 +217,41 @@ class EnrichmentFacility : public cyclus::Facility {
+  
+  inline void feed_commod(std::string in_com) { feed = in_com; }
+
+  inline std::string feed_commod() const { return feed; }
```
gcsg, getters/setters for a variable `foo` should be `get_foo` and `set_foo`, if you update the name `feed` you will not have to change these

gidden commented on src/enrichment_facility.h:
```diff
@@ -186,40 +217,41 @@ class EnrichmentFacility : public cyclus::Facility {
+  inline std::string tails_commodity() const { return tails_commod; }
 
-  inline void InRecipe(std::string in_rec) { in_recipe = in_rec; }
+  inline void InRecipe(std::string in_rec) { feed_recipe = in_rec; }
```
gcsg function name

gidden commented on src/enrichment_facility.h:
```diff
@@ -186,40 +217,41 @@ class EnrichmentFacility : public cyclus::Facility {
   }
 
-  inline std::string out_commodity() const { return out_commod; }
+  inline std::string tails_commodity() const { return tails_commod; }
```
gcsg function name

gidden commented on src/enrichment_facility.h:
```diff
@@ -186,40 +217,41 @@ class EnrichmentFacility : public cyclus::Facility {
 
-  inline void out_commodity(std::string out_com) {
-    out_commod = out_com;
+  inline void tails_commodity(std::string tails_com) {
```
gcsg function name

gidden commented on src/enrichment_facility.h:
```diff
@@ -186,40 +217,41 @@ class EnrichmentFacility : public cyclus::Facility {
+  }
 
-  inline std::string in_commodity() const { return in_commod; }
+  inline std::string product_commod() const { return product; }
```
gcsg function name

gidden commented on src/enrichment_facility.h:
```diff
@@ -186,40 +217,41 @@ class EnrichmentFacility : public cyclus::Facility {
+  inline void InRecipe(std::string in_rec) { feed_recipe = in_rec; }
 
-  inline std::string InRecipe() const { return in_recipe; }
+  inline std::string InRecipe() const { return feed_recipe; }
```
gcsg function name

gidden commented on src/enrichment_facility.h:
```diff
@@ -233,81 +265,127 @@ class EnrichmentFacility : public cyclus::Facility {
 
   inline double CurrentSwuCapacity() const { return current_swu_capacity; }
 
+  inline void MaxEnrich(double enrichment) { max_enrich = enrichment; }
```
gcsg function name

gidden replied:
it looks like there are a bunch of gcsg function name issues. can you address them each please and ping when ready?

mbmcgarry commented:
Ping - the EnrichmentFacility has been updated again.  I have addressed the comments from the previous review regarding:
- changed state vars feed, product to feed_commod, product_commod
- removed setter/getter functions to address gcsg function name issues in a way consistent with other new facilities.
- found typo that was likely causing nosetests to fail
- fixed formatting issues

More generally, the new Enrichment facility now includes these features as well:
- Added preference adjustments to select  for input material with higher concentrations of U235 (not to exceed the assay of the output bid). This feature is on by default and can be turned off with the order_prefs flag.
- Relaxed the input material constraint so that materials are accepted even if they do not exactly match the EF's requested recipe.
- Instituted a check so that input material is not accepted if it has no U235- Changed the way FeedQty is calculated to account for non-U235,U-238 contributions to the material
- Added a maximum enrichment variable - if set then the enrichment facility will not fulfill any bids for material with higher enrichments. Includes checks using custom XML schema to be sure that user-input max_enrich is between 0 and 1.
- Added a tails commodity to allow trading of tails with other agents
- Changed inherited state variable names to better match the design document specifications (ie, clarifying capacity, inventory, thruput, etc). Updated input files.
- Wrote new tests for these features using MockSim
- Replaced ResourceBuf with ResBuf
-Changed facility name to Enrichment
- Cleaned up old tests

Unfortunately, nosetests is completely unable to run on mac currently, so someone else will need to confirm that this passes.

gidden commented:
I still see the test failure
```
======================================================================
FAIL: test_deploy (test_regression.TestPhysorEnrichment)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/gidden/work/cyclus/cycamore/tests/test_regression.py", line 77, in test_deploy
    assert_equal(len(self.enr_id), 1)
AssertionError: 0 != 1
-------------------- >> begin captured stdout << ---------------------
TESTING: running command in /home/gidden/work/cyclus/cycamore/tests:

cyclus -o 169ea9b9-bf4f-49f3-9696-87112663a1e1.h5 --input-file ../input/physor/1_Enrichment_2_Reactor.xml


--------------------- >> end captured stdout << ----------------------

----------------------------------------------------------------------
Ran 13 tests in 2.147s

FAILED (failures=1)
```

gidden commented:
Hi @mbmcgarry, this looks really good. Can you provide some short justification for the tests you removed?

mbmcgarry commented on src/enrichment_facility_tests.cc:
```diff
@@ -11,229 +11,484 @@
-  EXPECT_DOUBLE_EQ(swu_capacity, src_facility->SwuCapacity());
-  EXPECT_EQ(reserves, src_facility->InitialReserves());
-}
-
```
Initialization of facility is implicitly tested by MockSim instances

rwcarlsen replied:
And, on top of what meghan said, this only tested getter/setter pairs anyway.

mbmcgarry commented on src/enrichment_facility_tests.cc:
```diff
@@ -11,229 +11,484 @@
-  EXPECT_DOUBLE_EQ(swu_capacity, fac.SwuCapacity());
-  EXPECT_EQ(reserves, fac.InitialReserves());
-}
-
```
Implicitly tested by MockSim instance

mbmcgarry commented on src/enrichment_facility_tests.cc:
```diff
@@ -11,229 +11,484 @@
-
-  delete cloned_fac;
-}
-
```
Implicitly tested by MockSim

rwcarlsen replied:
And since this is a code-generated cloning function,  we shouldn't be very concerned about testing it.

mbmcgarry commented on src/enrichment_facility_tests.cc:
```diff
@@ -11,229 +11,484 @@
-  EXPECT_NO_THROW(DoAddMat(GetMat(inv_size)));
-  EXPECT_THROW(DoAddMat(GetMat(1)), cyclus::Error);
-}
-
```
Implicitly tested by MockSim

mbmcgarry commented on src/enrichment_facility_tests.cc:
```diff
@@ -11,229 +11,484 @@
-  EXPECT_DOUBLE_EQ(q.atom_frac(922380000), u238 / (u235 + u238));
-  EXPECT_DOUBLE_EQ(mat->quantity(), qty);
-}
-
```
material composition and quantity is covered in new test RequestEnrich

mbmcgarry commented on src/enrichment_facility_tests.cc:
```diff
@@ -259,175 +515,10 @@ TEST_F(EnrichmentFacilityTest, ValidReq) {
-  ports = src_facility->GetMatlRequests();
-  EXPECT_TRUE(ports.empty());
-}
-
```
This tested EnrichmentFacility function GetMatlRequests, which is implicitly tested in MockSim instance

mbmcgarry commented on src/enrichment_facility_tests.cc:
```diff
@@ -259,175 +515,10 @@ TEST_F(EnrichmentFacilityTest, ValidReq) {
-      ports.begin()->get()->constraints();
-  EXPECT_EQ(constraints.size(), 0);
-}
-
```
This tested that a valid request was traded - replaced with RequestQty test.

mbmcgarry commented on src/enrichment_facility_tests.cc:
```diff
@@ -259,175 +515,10 @@ TEST_F(EnrichmentFacilityTest, ValidReq) {
-      5, cyclus::Composition::CreateFromMass(v));
-  Material::Ptr response = base->ExtractComp(6, target->comp());
-}
-
```
This test didn't test anything.

mbmcgarry commented on src/enrichment_facility_tests.cc:
```diff
@@ -259,175 +515,10 @@ TEST_F(EnrichmentFacilityTest, ValidReq) {
-  delete bid1;
-  delete req2;
-  delete req1;
-}
```
Tested that incoming material was accurately added to inventory - replaced with RequestQty

gidden commented:
@mbmcgarry, can you rebase this on develop and update the test please? it should be ready to go after that.

mbmcgarry commented on src/enrichment_facility_tests.cc:
```diff
@@ -259,175 +515,10 @@ TEST_F(EnrichmentFacilityTest, ValidReq) {
-  EXPECT_TRUE(*constrs.begin() == swu || *(++constrs.begin()) == swu);
-  EXPECT_TRUE(*constrs.begin() == natu || *(++constrs.begin()) == natu);
-}
-
```
Tested that constraints were properly added to EF bids - implicitly tested by MockSim in the new tests CheckSWUConstraint and CheckCapConstraint

mbmcgarry commented on src/enrichment_facility_tests.cc:
```diff
@@ -546,57 +642,37 @@ TEST_F(EnrichmentFacilityTest, Response) {
-  EXPECT_TRUE(cyclus::AlmostEq(src_facility->CurrentSwuCapacity(), 0));
-
-  // too much qty, capn!
-  trade = Trade<Material>(req, bid, 1);  // a small number
```
SWU now tested CheckSWUConstraint

mbmcgarry commented on src/enrichment_facility_tests.cc:
```diff
@@ -259,175 +515,10 @@ TEST_F(EnrichmentFacilityTest, ValidReq) {
-  }
-  return ec;
-}
-
```
This fn is no longer used by any tests.

mbmcgarry commented:
See inline comments for deleted tests - most were replaced by new tests using MockSim. If I missed explaining any deleted functionality, let me know and I can point you to the section of replacement code.


On Apr 9, 2015, at 1:28 PM, Matthew Gidden <notifications@github.com<mailto:notifications@github.com>> wrote:


Hi @mbmcgarry<https://github.com/mbmcgarry>, this looks really good. Can you provide some short justification for the tests you removed?

—
Reply to this email directly or view it on GitHub<https://github.com/cyclus/cycamore/pull/305#issuecomment-91319282>.

gidden commented:
thanks!

On Thu, Apr 9, 2015 at 4:40 PM, mbmcgarry <notifications@github.com> wrote:

> See inline comments for deleted tests - most were replaced by new tests
> using MockSim. If I missed explaining any deleted functionality, let me
> know and I can point you to the section of replacement code.
>
>
> On Apr 9, 2015, at 1:28 PM, Matthew Gidden <notifications@github.com
> <mailto:notifications@github.com>> wrote:
>
>
> Hi @mbmcgarry<https://github.com/mbmcgarry>, this looks really good. Can
> you provide some short justification for the tests you removed?
>
> —
> Reply to this email directly or view it on GitHub<
> https://github.com/cyclus/cycamore/pull/305#issuecomment-91319282>.
>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cyclus/cycamore/pull/305#issuecomment-91363733>.
>



-- 
Matthew Gidden, Ph.D.
Postdoctoral Associate, Nuclear Engineering
The University of Wisconsin -- Madison
Ph. 225.892.3192

rwcarlsen commented:
@gidden - Meghan and I sat down together a while back and went through the old tests, pruning and adding new ones.  So you can flog me if anything is broken (that would have been caught by deleted tests :)

gidden commented:
Future potential flogging noted =). I just wanted a quick reason for each,
they all sound quite reasonable.

On Thu, Apr 9, 2015 at 5:05 PM, Robert Carlsen <notifications@github.com>
wrote:

> @gidden <https://github.com/gidden> - Meghan and I sat down together a
> while back and went through the old tests, pruning and adding new ones. So
> you can flog me if anything is broken (that would have been caught by
> deleted tests :)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cyclus/cycamore/pull/305#issuecomment-91368045>.
>



-- 
Matthew Gidden, Ph.D.
Postdoctoral Associate, Nuclear Engineering
The University of Wisconsin -- Madison
Ph. 225.892.3192

gidden commented:
Just to note explicitly, this will change a user-facing API in a non-experimental archetype. @rwcarlsen should probably sign off on this as cycamore manager.

mbmcgarry commented:
Rebased.

gidden commented:
this passes for me. waiting for @rwcarlsen to confirm.

rwcarlsen commented:
We are a go.

On Fri, Apr 10, 2015 at 10:25 AM, Matthew Gidden <notifications@github.com>
wrote:

> this passes for me. waiting for @rwcarlsen <https://github.com/rwcarlsen>
> to confirm.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cyclus/cycamore/pull/305#issuecomment-91592199>.
>

gidden commented:
Thanks @mbmcgarry! Great work!

mbmcgarry commented:
Woohoo! Thanks to all for helping me learn about cyclus and archetype development!


On Apr 10, 2015, at 12:11 PM, Matthew Gidden <notifications@github.com<mailto:notifications@github.com>> wrote:


Thanks @mbmcgarry<https://github.com/mbmcgarry>! Great work!

—
Reply to this email directly or view it on GitHub<https://github.com/cyclus/cycamore/pull/305#issuecomment-91624290>.
